python/tests/test_namedlock.py

```diff
@@ -0,0 +1,117 @@
+import unittest
+import tempfile
+import threading
+import time
+from pathlib import Path
+import concurrent.futures
+
+from bucketbase.namedlock import ThreadLockManager, FileLockManager
+
+
+class ThreadLockManagerTests(unittest.TestCase):
+    def setUp(self):
+        self.lock_manager = ThreadLockManager()
+
+    def test_get_lock_returns_same_lock_for_same_name(self):
+        lock1 = self.lock_manager.get_lock("test_lock")
+        lock2 = self.lock_manager.get_lock("test_lock")
+        self.assertIs(lock1, lock2)
+
+    def test_get_lock_returns_different_locks_for_different_names(self):
+        lock1 = self.lock_manager.get_lock("test_lock1")
+        lock2 = self.lock_manager.get_lock("test_lock2")
+        self.assertIsNot(lock1, lock2)
+
+    def test_lock_actually_locks(self):
```

Comment:
This one seems redundant, since it tests if the existing threading.Lock actually works.
It wouldn't hurt, if it wouldn't take so much time to run. 
Be aware - the unit-tests should be lightning fast! Tests running time is also very very important on big projects (for some repositories, inefficient tests costs tens of millions of $)